### PR TITLE
Fix event path trimming test case

### PR DESCRIPTION
**Pull request by TakayoshiKochi**
Event retargeting happens at the shadow tree boundary and event
handler should be installed above the shadow root.

In the old test the handler is installed on the shadow root, which
retargeting is not yet happening while bubbling the event.

Also change the initial target and relatedTarget to be different to
make sure if they are retargeted to the same node.

### CODE REVIEW

**Comment by TakayoshiKochi** [comment]
@hayatoito PTAL

**Comment by TakayoshiKochi** [comment]
This is to the response to your comment on #2097.

File: shadow-dom/events/event-dispatch/test-003.html [hunk 1]
```diff
@@ -37,19 +37,29 @@
...
+    var div2 = d.createElement('div');
+    div2.setAttribute('style', 'height:50%; width:100%');
+    div2.setAttribute('id', 'div2');
+    s.appendChild(div2);
+
+    // Add an event listner above its shadow root. If both target and
+    // relatedTarget are in the same shadow tree under the root,
+    // retargeted target and relatedTarget will be the same node
+    // and the event must not invoke the handler.
+    host.addEventListener('mouseover', A_05_05_03_T01.step_func(function(event) {
```
Comment (hayatoito) [inline shadow-dom/events/event-dispatch/test-003.html]: An event listener should be invoked by a trusted event, shouldn't it?

Comment (TakayoshiKochi) [inline shadow-dom/events/event-dispatch/test-003.html]: IIUC a trusted event must not invoke an event, if target and relatedTarget are same.
This test dispatches a "mouseover" event whose target is div1 and relatedTarget is div2,
and div1 and div2 are in the same TreeScope.  So once the event bubbles up out of the
TreeScope, target and relatedTarget become same and will not invoke the event handler.

Or are you saying we should dispatch "trusted event" for this test?
I don't think we have interoperable way to dispatch a trusted event in web platform tests.

Comment (hayatoito) [inline shadow-dom/events/event-dispatch/test-003.html]: An event created by a script, e.g. document.createEvent("MouseEvents"), isn't a trusted event, I think.

> I don't think we have interoperable way to dispatch a trusted event in web platform tests.

That's the point. I'm afraid that we don't have a way to test this feature.

**Comment by TakayoshiKochi** [comment]
Now I noticed 'focusin' event could have both `target` and `relatedTaget`, and hopefully
`focus()` can fire trusted 'focusin' events.
I modified the assertion so that if the user agent has `isTrusted` property on event object,
we check it strictly, otherwise we check and show different failure message.

@hayatoito PTAL

**Comment by hayatoito** [comment]
Hmm. I think we should write a test for one well-defined behavior.
I'm afraid that this test would hide ambiguous behavior. It's hard to read what's expected behavior from this test. 

e.g.  'focusin' is a trusted event in this test? Reader can't tell it easily...

**Comment by TakayoshiKochi** [comment]
I discussed in #whatwg IRC with @annevk, smaug, @Ms2ger and conclusion was that
focus event generated by user agent, that are result of moving focus from one element to
another is **trusted**.  So the result of `focus()` is also trusted.

For example,
[html/editing/focus/focus-management/focus-events.html](https://github.com/w3c/web-platform-tests/blob/master/html/editing/focus/focus-management/focus-events.html)
does `focus()` and checks if `isTrusted` is `true` in `onfocus` handler. 

In http://www.w3.org/TR/html51/editing.html#processing-model-6
> When a user agent is required to fire a focus event named e at an element t and with a given related target r, the user agent must create a trusted FocusEvent object, 
which is the definitive reference.

**Comment by hayatoito** [comment]
Thus, if we can assume that the `focusin` event is trusted here, I think you can replace L54-L63 with one-line, `assert_true(false, "xxx")`, for the purpose of this test. :)

However, I think there is more better way to test an event path trimming. Why don't you use `event.deepPath`? That's more reliable, I think.

**Comment by TakayoshiKochi** [comment]
@hayatoito PTAL

Changed to use .deepPath.

**Comment by TakayoshiKochi** [comment]
@hayatoito ping?

**Comment by hayatoito** [comment]
Could you squash your commits into one commit?

**Comment by TakayoshiKochi** [comment]
Created #2116.

**Pull request closed without merging**